crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -12,94 +12,143 @@ pub use bevy_ecs_macros::WorldQuery;
+/// - **[`Entity`].**
+///   Gets the identifier of the queried entity.
+/// - **[`Option`].**
+///   By default, a world query only tests entities that are compatible with its type.
+///   Wrapping it into an `Option` will increase the query search space, and it will return `None` if an entity doesn't satisfy the `WorldQuery`.
+/// - **[`AnyOf`].**
+///   Equivalent to wrapping each world query inside it into an `Option`.
+/// - **[`ChangeTrackers`].**
+///   Similar to change detection filters but it is used as a query fetch parameter.
+///   It exposes methods to check for changes to the wrapped component.
 ///
-/// Notable types that implement this trait are `&T` and `&mut T` where `T` implements [`Component`],
-/// allowing you to query for components immutably and mutably accordingly.
-///
-/// See [`Query`](crate::system::Query) for a primer on queries.
-///
-/// # Basic [`WorldQuery`]'s
-///
-/// Here is a small list of the most important world queries to know about where `C` stands for a
-/// [`Component`] and `WQ` stands for a [`WorldQuery`]:
-/// - `&C`: Queries immutably for the component `C`
-/// - `&mut C`: Queries mutably for the component `C`
-/// - `Option<WQ>`: Queries the inner [`WorldQuery`] `WQ` but instead of discarding the entity if the world
-///     query fails it returns [`None`]. See [`Query`](crate::system::Query).
-/// - `(WQ1, WQ2, ...)`: Queries all contained world queries allowing to query for more than one thing.
-///     This is the `And` operator for filters. See [`Or`].
-/// - `ChangeTrackers<C>`: See the docs of [`ChangeTrackers`].
-/// - [`Entity`]: Using the entity type as a world query will grant access to the entity that is
-///     being queried for. See [`Entity`].
-///
-/// Bevy also offers a few filters like [`Added`](crate::query::Added), [`Changed`](crate::query::Changed),
-/// [`With`](crate::query::With), [`Without`](crate::query::Without) and [`Or`].
-/// For more information on these consult the item's corresponding documentation.
-///
-/// [`Or`]: crate::query::Or
-///
-/// # Derive
+/// Implementing the trait manually can allow for a fundamentally new type of behaviour.
 ///
-/// This trait can be derived with the [`derive@super::WorldQuery`] macro.
+/// # Safety
 ///
-/// You may want to implement a custom query with the derive macro for the following reasons:
-/// - Named structs can be clearer and easier to use than complex query tuples. Access via struct
-///   fields is more convenient than destructuring tuples or accessing them via `q.0, q.1, ...`
-///   pattern and saves a lot of maintenance burden when adding or removing components.
-/// - Nested queries enable the composition pattern and makes query types easier to re-use.
-/// - You can bypass the limit of 15 components that exists for query tuples.
+/// component access of `ROQueryFetch<Self>` must be a subset of `QueryFetch<Self>`
+/// and `ROQueryFetch<Self>` must match exactly the same archetypes/tables as `QueryFetch<Self>`
```

Comment:
This safety comment is already at the bottom of this doc comment so can be removed